### ISSUE TITLE
Global import

### ISSUE THREAD

**Issue by lucas404x** [dart-lang/sdk#50053]
Is there any plan to implement something like a `global using` on dart?
In csharp, we can create a separate file and declare global libraries. It avoids a lot of imports in several files.

I think it will be helpful when you use custom extensions because you need to type the whole method first to the auto import works well.

**Comment by eernstg** [comment]
There is no such mechanism in a narrow sense.

(I'll move this issue to the language repository because it amounts to a request for this kind of feature.)

The normal way to get a similar effect in Dart is to have a library that exports many other libraries, e.g., like https://github.com/dart-lang/intl/blob/5464b3b95635e97f67e29d71bdbe85a5d4126971/lib/intl.dart#L21.

For any particular software artifact that you are maintaining (e.g., a package), you could choose to maintain a library `globals.dart` that serves this specific purpose and does nothing other than exporting all those libraries that you wish to make "global". Note that you can use `hide` to refine the namespace offered by `globals.dart`.

Anyone who wishes to have a approximately the same namespace could export `globals.dart`, remove some names using `hide`, and add other names using additional exports.

In other words, the existing export mechanism allows for the provision or large namespaces in a way which is rather straightforward and also customizable.

It's certainly possible that a notion of namespaces and global visibility could be a useful addition to Dart, but I also think that it's useful to compare with the existing mechanism. In particular, it's useful to consider how we'd handle conflicts that arise because a newly introduced declaration in one of those "global" libraries based on namespaces.
